### ISSUE TITLE
Unable to connect until next restart after bad handshake

### ISSUE THREAD

**Issue by Quantumplation** [txpipe/dolos#292]
If you attempt to connect via the dolos socket with an incorrect network magic, you will get an `HandshakeProtocol(Plexer(AgentDequeue))` error.

However, even after correcting the network magic, subsequent connections will fail with connection refused:

```
ConnectFailure(Os { code: 61, kind: ConnectionRefused, message: "Connection refused" })
```

**Comment by Quantumplation** [comment]
Note that, much like #293, the only indication that something is wrong comes when you shut down the node

```
2024-07-13T20:46:59.257467Z  INFO stage{stage="ledger"}: gasket::runtime: switching stage phase prev_phase=Teardown next_phase=Ended
Error:   × serve thread
  ├─▶ serving Ouroboros
  ╰─▶ server error: handshake version not accepted
```

**Comment by scarmuega** [comment]
fixed by #297
